uaclient/files/user_config_file.py

```diff
@@ -119,14 +125,15 @@ def redact_config_data(
                     )
         return redacted_data
 
-    def read(self) -> Optional[UserConfigData]:
-        try:
-            if util.we_are_currently_root():
-                return self._private.read()
-            return self._public.read()
-        except Exception as e:
-            LOG.warning("Error reading user config", exc_info=e)
-            return None
+    def read(self) -> UserConfigData:
+        if util.we_are_currently_root():
+            private_config = self._private.read()
+            if private_config is not None:
+                return private_config
+        public_config = self._public.read()
+        if public_config is not None:
+            return public_config
+        return UserConfigData()
```

Comment:
Why are we returning an empty `UserConfigData` object here ? Is it just for us to have the default values in place ?

Reply:
Yes, to have the default values in place in case there are no files to read so the Config always has a user_config set.